### ISSUE TITLE
Error when running meteor publish --update

### ISSUE THREAD

**Issue by copleykj** [meteor/meteor#13394]
The following error occurs when running `meteor publish --update`. OS is Linux.

```sh
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string or an i
nstance of Buffer or URL. Received an instance of Promise
    at Object.statSync (node:fs:1664:10)
    at Object.stat (/home/copleykj/.meteor/packages/meteor-tool/.3.0.3.sq73dw.qasz++
os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/
fs/tools/fs/files.ts:1520:23)
    at uploadFile (/tools/packaging/package-client.js:330:20)
    at /tools/packaging/package-client.js:546:16
    at Object.enterJob (/tools/utils/buildmessage.js:387:12)
    at Object.exports.updatePackageMetadata (/tools/packaging/package-client.js:539:
3)
    at /tools/cli/commands-packages.js:208:9
    at Object.capture (/tools/utils/buildmessage.js:282:5)
    at Object.main.captureAndExit (/tools/cli/main.js:273:16)
    at updatePackageMetadata (/tools/cli/commands-packages.js:204:5)
    at Command.func (/tools/cli/commands-packages.js:358:12)
    at /tools/cli/main.js:1546:15 {
  code: 'ERR_INVALID_ARG_TYPE'
```

**Comment by nachocodoner** [comment]
We've backlogged this issue. We've encountered some problems with the involved Meteor app and async adaptation, but we'll review and fix it.

Thanks for reporting

**Comment by Grubba27** [comment]
Hello! I'm taking on this.

from my first diagnosis check, I think I've found where the issue is. Based on your trace:

```text
  at Object.exports.updatePackageMetadata (/tools/packaging/package-client.js:539:
3)
    at /tools/cli/commands-packages.js:208:9
    at Object.capture (/tools/utils/buildmessage.js:282:5)
    at Object.main.captureAndExit (/tools/cli/main.js:273:16)
    at updatePackageMetadata (/tools/cli/commands-packages.js:204:5)
```

It leads me here:

https://github.com/meteor/meteor/blob/f1bb0ea9f9aa628d9ff8a8980a01223899b7eb23/tools/packaging/package-client.js#L539-L551

The snippet itself is not helpful, but when we look inside my IDE(TypeScript saved the day!) and with the help of your error: `The "path" argument must be of type string or an instance of Buffer or URL. Received an instance of Promise` we can find the error: 

<img width="679" alt="Screenshot 2024-11-13 at 19 22 30" src="https://github.com/user-attachments/assets/b56cb0d6-fc31-44f7-982a-b8c617c1eadc">

The other instance of this function [`saveReadmeToTmp`](https://github.com/meteor/meteor/blob/f1bb0ea9f9aa628d9ff8a8980a01223899b7eb23/tools/packaging/package-client.js#L666) is already using async correctly, it must have been forgotten to be checked when we turned it into async.


I'll try making a test for it and pull a PR soon
